staging/src/k8s.io/kube-apiextensions-server/test/integration/basic_test.go

```diff
@@ -167,3 +182,97 @@ func TestSimpleCRUD(t *testing.T) {
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// check whether it shows up in discovery properly
+	resources, err := apiExtensionClient.Discovery().ServerResourcesForGroupVersion(group + "/" + version)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(resources.APIResources) != 1 {
+		t.Fatalf("Expected exactly the resource \"noxus\" in group version %v/%v via discovery, got: %v", group, version, resources.APIResources)
+	}
+
+	r := resources.APIResources[0]
+	if r.Name != "noxus" {
+		t.Fatalf("Expected exactly the resource \"noxus\" in group version %v/%v via discovery, got: %v", group, version, r.Name)
+	}
+	if r.Kind != "WishIHadChosenNoxu" {
+		t.Fatalf("Expected exactly the kind \"WishIHadChosenNoxu\" in group version %v/%v via discovery, got: %v", group, version, r.Kind)
+	}
+
+	s := []string{"foo", "bar", "abc", "def"}
+	if !reflect.DeepEqual(r.ShortNames, s) {
+		t.Fatalf("Expected exactly the shortnames `foo, bar, abc, def` in group version %v/%v via discovery, got: %v", group, version, r.ShortNames)
+	}
+
+	sort.Strings(r.Verbs)
+	expectedVerbs := []string{"create", "delete", "deletecollection", "get", "list", "patch", "update", "watch"}
+	if !reflect.DeepEqual([]string(r.Verbs), expectedVerbs) {
+		t.Fatalf("Unexpected verbs for resource \"noxus\" in group version %v/%v via discovery: expected=%v got=%v", group, version, expectedVerbs, r.Verbs)
+	}
+}
+
+func TestNonNamespaceReject(t *testing.T) {
+	stopCh, apiExtensionClient, clientPool, err := testserver.StartDefaultServer()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer close(stopCh)
+
+	noxuDefinition := testserver.NewNoxuCustomResourceDefinition(apiextensionsv1alpha1.NamespaceScoped)
+	noxuVersionClient, err := testserver.CreateNewCustomResourceDefinition(noxuDefinition, apiExtensionClient, clientPool)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ns := ""
+	noxuResourceClient := NewNamespacedCustomResourceClient(ns, noxuVersionClient, noxuDefinition)
+	initialList, err := noxuResourceClient.List(metav1.ListOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if e, a := 0, len(initialList.(*unstructured.UnstructuredList).Items); e != a {
+		t.Errorf("expected %v, got %v", e, a)
+	}
+	initialListTypeMeta, err := meta.TypeAccessor(initialList)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if e, a := noxuDefinition.Spec.Group+"/"+noxuDefinition.Spec.Version, initialListTypeMeta.GetAPIVersion(); e != a {
+		t.Errorf("expected %v, got %v", e, a)
+	}
+	if e, a := noxuDefinition.Spec.Names.ListKind, initialListTypeMeta.GetKind(); e != a {
+		t.Errorf("expected %v, got %v", e, a)
+	}
+
+	createdNoxuInstance, err := instantiateCustomResource(t, testserver.NewNoxuInstance(ns, "foo"), noxuResourceClient, noxuDefinition)
+	if err == nil {
+		t.Fatalf("unexpected non-error while creating noxu instance: %v", createdNoxuInstance)
+	}
+}
+
+func TestSameNameDiffNamespace(t *testing.T) {
+	scope := apiextensionsv1alpha1.NamespaceScoped
+	ns1 := "namespace-1"
+	testSimpleCRUD(t, scope, ns1)
```

Comment:
each of these creates its own server, doesn't it? We have to share it though for this test.